src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -58,7 +62,19 @@ internal sealed partial class TlsOverPerCoreLockedStacksArrayPool<T> : ArrayPool
         [ThreadStatic]
         private static Bucket[] t_tlsBuckets;
 
-        private static ConditionalWeakTable<Bucket[], object> s_allTlsBuckets = new ConditionalWeakTable<Bucket[], object>();
+        private int _callbackCreated;
+
+        /// <summary>
+        /// Used to keep track of all thread local buckets for trimming if needed
+        /// </summary>
+        private static readonly ConditionalWeakTable<Bucket[], object> s_AllTlsBuckets;
+
+        private unsafe static int s_typeSize = Unsafe.SizeOf<T>();
+
+        static TlsOverPerCoreLockedStacksArrayPool()
```

Comment:
Explicit static constructors makes the type non-before field init. It tends to have pretty bad performance consequences.

Reply:
Any suggestions on getting ordering correct? The kill switch wasn't initialized when I tried to hit it in line.

Reply:
Do not use the fancy syntax, and just use the 

```
static bool s_TrimBuffers = GetTrimBuffers();
static ConditionalWeakTable<...> s_AllTlsBuckets =  s_TrimBuffers ? GetTrimBuffers() : null;
```